apps/web/src/app/[locale]/challenge/[slug]/solutions/_components/solution-detail.tsx

```diff
@@ -51,18 +48,17 @@ export function SolutionDetails({ solution }: Props) {
       });
     }
   };
-
   return (
     <div className="relative h-full">
       <div className="relative flex h-full flex-col">
         <div className="custom-scrollable-element flex-1 overflow-y-auto px-4 pb-16 pt-3">
           <div className="mb-5 flex flex-col gap-3">
             <div className="flex items-center gap-3">
-              <button onClick={handleClickLeftIcon} className="h-7 w-7">
+              <Link href={`/challenge/${slug}/solutions?${queryString}`} className="h-7 w-7">
```

Comment:
I add querystring 
because without querystring users can't return to the page they were viewing.

without querystring
1. challenge/get-return-type/solutions?page=3
2. challenge/get-return-type/solutions/49
3. challenge/get-return-type/solutions

The behavior I expected with querystring
1. challenge/get-return-type/solutions?page=3
2. challenge/get-return-type/solutions/49?page=3
3. challenge/get-return-type/solutions?page=3